feat(hu): add Hungarian locale

MMate2007 opened the pull request:
This PR adds initial Hungarian localisation based especially on the Hungarian Roman Missal (1991, editio typica altera), secondarily on the Hungarian [_Ordo >

Only those celebrations names were localised which are celebration in the national Hungarian calendar, however, no missing celebrations were added (those nee>

fix #248

MMate2007 committed:
Create hu.ts

MMate2007 committed:
Update hu.ts

MMate2007 committed:
Update hu.ts

MMate2007 committed:
Update hu.ts

MMate2007 committed:
Update hu.ts

MMate2007 committed:
Edit hu.ts

MMate2007 committed:
Comment unlocalised strings.

tukusejssirs commented on lib/locales/hu.ts:
```diff
@@ -0,0 +1,938 @@
+    pelagius_of_cordoba_martyr: 'Saint Pelagius of Córdoba, Martyr',
+    pentecost_sunday: 'Pünkösdvasárnap',
+    perpetua_of_carthage_and_felicity_of_carthage_martyrs: 'Saints Perpetua and Felicity, Martyrs',*/
+    peter_and_paul_apostles: 'Szent Péter és Pál apostolok',
```
According to [_Direktórium 2022_](http://tar.liturgia.hu/Naptar/DIR2022_JAV_2021-12-13.pdf#page=124), it should be `Szent Péter és szent Pál apostolok` (I am not sure about the casing of `szent Pál` vs `Szent Pál` and `apostolok` vs `Apostolok` though). However, the Hungarian translation of the Roman Missal is authoritative and primary source, not the Ordo thus I am not sure if this is actually correct either.

tukusejssirs commented on lib/locales/hu.ts:
```diff
@@ -0,0 +1,938 @@
+    stanislaus_of_szczepanow_bishop: 'Saint Stanislaus, Bishop and Martyr',
+    stanislaus_of_szczepanow_bishop_patron_of_poland: 'Saint Stanislaus, Bishop and Martyr, Patron of Poland',
+    stephen_i_of_hungary: 'Saint Stephen of Hungary',*/
+    stephen_the_first_martyr: 'Szent István első vértanú',
```
According to [_Direktórium 2022_](http://tar.liturgia.hu/Naptar/DIR2022_JAV_2021-12-13.pdf#page=42), it should be `Szent István diakónus, első vértanú`. Again, if it is different from the Hungarian Roman Missal, we should prefer the one from the Roman Missal.

tukusejssirs commented on lib/locales/hu.ts:
```diff
@@ -0,0 +1,938 @@
+    stanislaus_of_szczepanow_bishop_patron_of_poland: 'Saint Stanislaus, Bishop and Martyr, Patron of Poland',
+    stephen_i_of_hungary: 'Saint Stephen of Hungary',*/
+    stephen_the_first_martyr: 'Szent István első vértanú',
+    sunday_of_the_word_of_god: 'Évközi III. vasárnap vagy az Isteni Ige vasárnapja',
```
According to [_Direktórium_](http://tar.liturgia.hu/Naptar/DIR2022_JAV_2021-12-13.pdf#page=55), it should be `Évközi III. vasárnap, vagy Isten Igéjenek vasárnapja`. I am not sure about the casing of `Igéjenek`, however I presume that it should be capitalised.

tukusejssirs commented on lib/locales/hu.ts:
```diff
@@ -0,0 +1,938 @@
+  names: {
+    //'205_blessed_martyrs_of_japan': '205 Blessed Martyrs of Japan',
+    adalbert_of_prague_bishop: 'Szent Adalbert püspök és vértanú',
+    adalbert_of_prague_bishop_patron_of_poland: 'Szent Adalbert püspök és vértanú, Legyelország patrónusa',
```
According to [_Direktórium 2022_](http://tar.liturgia.hu/Naptar/DIR2022_JAV_2021-12-13.pdf#page=163), in Hungarian, you use _(fő-) pátrónája_ instead of `patrónusa`.

Just a note: in the key name (e.g. `adalbert_of_prague_bishop_patron_of_poland`), we use `patron`/`patroness` for principal patrons (`főpátrón`) and `copatron`/`copatroness` for `??? másodlagos pátrón` (I am not sure if _másodlagos_ is the word for secondary patron in Hungarian).

This should be changed (unless the Hungarian Roman Missal contradicts this) in all places, not just in this key.

tukusejssirs commented on lib/locales/hu.ts:
```diff
@@ -0,0 +1,938 @@
+    germanus_of_auxerre_bishop: 'Saint Germanus of Auxerre, Bishop',
+    gertrude_of_nivelles_abbess: 'Saint Gertrude of Nivelles, Abbess',
+    gertrude_the_great_virgin: 'Saint Gertrude, Virgin',*/
+    gisela_of_hungary: 'Boldog Gizella',
```
According to [_Direktórium 2022_](http://tar.liturgia.hu/Naptar/DIR2022_JAV_2021-12-13.pdf#page=102), it should be `Boldog Gizella királyné`.

tukusejssirs commented on lib/locales/hu.ts:
```diff
@@ -0,0 +1,938 @@
+    gregory_i_the_great_pope: 'Saint Gregory the Great, Pope and Doctor of the Church',
+    gregory_of_narek_abbot: 'Saint Gregory of Narek, Abbot and Doctor of the Church',
+    gregory_vii_pope: 'Saint Gregory VII, Pope',*/
+    guardian_angel_of_portugal: 'Portugália őrangyala',
```
I am not sure if this is correct. According to [_Direktórium 2022_](http://tar.liturgia.hu/Naptar/DIR2022_JAV_2021-12-13.pdf#page=161), `holy_guardian_angels` are translated as `Szent Őrzőangyalok`, therefore I would translated this as `Portugália Őrzőangyala`, however, I might be wrong (I might be not aware of some Hungarian grammar rules), because when I searched the web, all I could find is `Portugália őrangyala`: I could find [39 results of `Portugália Őrzőangyala`](https://www.google.com/search?client=firefox-b-d&q=%22Portug%C3%A1lia%2B%C5%90rz%C5%91angyala%22+site%3A.hu) and [71 results of `Portugália Őrangyala`](https://www.google.com/search?client=firefox-b-d&q=%22Portug%C3%A1lia%2B%C5%90rangyala%22+site%3A.hu).

MMate2007 replied:
Well, I have no idea which is right.

tukusejssirs replied:
What is correct in Hungarian in general: _őrzőangyal_ or _őrangyal_?

MMate2007 replied:
Both are correct.

tukusejssirs replied:
If both are correct, then I suggest to use `Őrzőangyal` in order to be consistent with the value of `holy_guardian_angels`.

MMate2007 replied:
Ok.

tukusejssirs commented on lib/locales/hu.ts:
```diff
@@ -0,0 +1,938 @@
+    hildegard_of_bingen_abbess: 'Saint Hildegard of Bingen, Abbess and Doctor of the Church',
+    holy_child_of_cebu: 'The Holy Child of Cebú',
+    holy_family_of_jesus_mary_and_joseph: 'The Holy Family of Jesus, Mary and Joseph',*/
+    holy_guardian_angels: 'A szent őrzőangyalok',
```
According to [Direktórium 2022](http://tar.liturgia.hu/Naptar/DIR2022_JAV_2021-12-13.pdf#page=161), there is no definite article (`a`) at the beginning of the celebration name. I have no idea if that is so in the Hungarian Roman Missal.

tukusejssirs reviewed (changes requested):
I have noticed some (potential) issues. Could you verify these issues with th Hungarian translation of the Roman Missal please? :wink:

I’ll change this PR to WIP (draft).

MMate2007 commented:
@tukusejssirs Thank you for your review! I'll correct my mistakes. Sorry about them. I wasn't sure if I could use Direktórium because of copyright laws.

tukusejssirs commented:
> Sorry about them.

Be sure there’ll be more. Even I’ve made some stupid mistake that were fixed after long time (even a year). :wink:

> I wasn't sure if I could use Direktórium because of copyright laws.

Well, as for the copyright, I am sure either, however, each celebration has a single official name at any one moment of time, thus I use the best source (priority order: primary, secondary, ternary source) available to me, therefore Ise Ordos/Directories as well.

MMate2007 committed:
Correcting issues according to Direktórium 2022

MMate2007 commented:
I corrected the errors (except one, where I don't know which one is right). I don't have access to the Hungarian Roman Missal now, so I couldn't check if those names are correct, but I think the names in the Direktórium are correct.

tukusejssirs commented:
> I corrected the errors (except one, where I don't know which one is right).

Thanks! :pray: You can mark my comments (in this PR) as resolved then. :wink:

> I don't have access to the Hungarian Roman Missal now

Just a suggestion: would it be a problem to ask the priest in your parish to let you take some pictures from the Roman Missal? It could be simply done, for example, in the sacristy after Holy Mass.

Note that you don’t have to do it, however, it would be very nice of view. Even the data from _Direktórium_ is good enough. :wink:

MMate2007 committed:
Update hu.ts

tukusejssirs commented on lib/locales/hu.ts:
```diff
@@ -0,0 +1,938 @@
+      season: 'Advent',
+      /*weekday: '$t(weekdays:{{dow}}, capitalize) of the $t(ordinals:{{week}}) week of Advent',
+      sunday: '$t(ordinals:{{week}}, capitalize) Sunday of Advent',
+      privileged_weekday: '$t(months:11, capitalize) {{day}}',*/
```
I suggest to use `//` (line comment) instead of `/* */` (block comment) to comment out unlocalised string in order to ease out uncommenting individual lines. YMMV. :wink:

tukusejssirs commented on lib/locales/hu.ts:
```diff
@@ -0,0 +1,938 @@
+    sunday_year_a: 'A év',
+    sunday_year_b: 'B év',
+    sunday_year_c: 'C év',
+    weekday_year_1: 'I. év',
+    weekday_year_2: 'II. év',
```
Are you sure that in Hungarian it is called _év_ (year)? In English and Slovak, it is called _cycle_ and _cyklus_ respectively, therefore I presume it is actually _ciklus_.

@emagnier, I think that `weekday_year_*` is a bit confusing, therefore I suggest to rename it to `weekday_cycle_*`. As for `sunday_year_*`, AFAIK both `year` and `cycle` are used, although I am not sure if either one of them is used consistently officially. Do you agree to rename `weekday_year_*` to `weekday_cycle_*`? I could do it if agree with the change.

MMate2007 replied:
In speech we use even and odd year but in the Reading book (I'm not sure how it is called in English) there is "I. évben" and "II. évben" near the readings.

tukusejssirs replied:
Even in Slovak (and I presume in English too) we use different names for the same thing (even/odd/first/second year/cycle), however, (at least in Slovak) in all official books we use _cyklus_ (cycle) consistently. Still, you might be right that in Hungarian _év_ is used for cycles, however, some proof would be very nice to have (e.g. look into _A szentmise olvasmányai hétköznapokra_ (ferial/weekday lectionary), volume 1 or 2, and you should be able to find that information somewhere on the first pages).

BTW, _Reading book_ is a lectionary in English.

MMate2007 replied:
Ok, I'll check it at the weekend.

emagnier replied:
> I think that `weekday_year_*` is a bit confusing, therefore I suggest to rename it to `weekday_cycle_*`. [...] I could do it if agree with the change.

@tukusejssirs: Indeed 👍! Go for it!

MMate2007 replied:
@tukusejssirs I've checked it and in the book there is *II. évben*.

tukusejssirs commented on lib/locales/hu.ts:
```diff
@@ -0,0 +1,938 @@
+    andre_grasset_priest: 'Blessed André Grasset, Priest and Martyr',*/
+    andrew_apostle: 'Szent András apostol',
+    andrew_apostle_patron_of_russia: 'Szent András apostol, Oroszország patrónusa',
+    andrew_apostle_patron_of_scotland: 'Szent András apostol, Skócia patrónusa',
```
Twice: `főpátrónája`; see [this comment](https://github.com/romcal/romcal/pull/403#discussion_r1014881440).

tukusejssirs commented on lib/locales/hu.ts:
```diff
@@ -0,0 +1,938 @@
+    francis_diaz_del_rincon_priest_and_companions_martyrs: 'Saint Francis Díaz del Rincon, Priest, and Companions, Martyrs',
+    francis_ferdinand_de_capillas_priest: 'Saint Francis Ferdinand de Capillas, Priest and Martyr',*/
+    francis_of_assisi: 'Assisi Szent Ferenc',
+    francis_of_assisi_patron_of_italy: 'Assisi Szent Ferenc, Olaszország patrónusa',
```
`főpátrónája`; see [this comment](https://github.com/romcal/romcal/pull/403#discussion_r1014881440).

tukusejssirs commented on lib/locales/hu.ts:
```diff
@@ -0,0 +1,938 @@
+    lelia_of_killeely_virgin: 'Saint Lelia, Virgin',
+    leo_i_the_great_pope: 'Saint Leo the Great, Pope and Doctor of the Church',
+    leo_ignatius_mangin_priest_and_companions_martyrs: 'Saint Leo Ignatius Mangin, Priest, and Companions, Martyrs',*/
+    leo_ix_pope: 'Szent IX Leó pápa',
```
Missing `.` after `IX`.

tukusejssirs commented on lib/locales/hu.ts:
```diff
@@ -0,0 +1,938 @@
+    ludger_of_munster_bishop: 'Saint Ludger of Münster, Bishop',
+    ludmila_of_bohemia_martyr: 'Saint Ludmila, Martyr',
+    luigi_orione_priest: 'Saint Luigi Orione, Priest',*/
+    luke_evangelist: 'Szent Lukács evangelista',
```
According to [_Direktórium 2022_](http://tar.liturgia.hu/Naptar/DIR2022_JAV_2021-12-13.pdf#page=166), you are missing an acute accent (`´`) → `evangélista`.

tukusejssirs commented on lib/locales/hu.ts:
```diff
@@ -0,0 +1,938 @@
+    rupert_of_salzburg_and_virgilius_of_salzburg_bishops: 'Saints Rupert and Virgilius of Salzburg, Bishops',
+    salomea_of_poland_religious: 'Blessed Salomea of Poland, Religious',
+    sancha_of_portugal_and_mafalda_of_portugal_virgins: 'Blessed Sancha and Mafalda, Virgins',*/
+    sara_salkahazi_virgin: 'Boldog Salkaházi Sára szűz és vértanú',
```
According to [Direktórium 2022](http://tar.liturgia.hu/Naptar/DIR2022_JAV_2021-12-13.pdf#page=104), there is a comma between `Sára` and `szűz`, however, it other celebration names, there is none. I personally miss the comma everywhere, however, if it is not used in the Hungarian translation of the Roman Missal, we should not use it anywhere.

In either case, we should be consistent: either use the comma everywhere or not, or find a rule when to use it and when not.

MMate2007 replied:
I've checked it in *A magyar helyesírás szabályai* and I found that there shouldn't be a comma between the name and the title. Well this is not completely clear for me, because it says that if the name has an appellative (like *vértanú*) there is no need for the comma. But if you can inflect the name and the appellative as well there should be a comma. However, in Hungarian we don't say ~~*Szent Máténak, apostolnak*~~, we say instead *Szent Máté apostolnak*, we don't inflect the name. I don't know if *Boldog Salkaházi Sárának, szűz és vértanúnak* is correct but I would say *Boldog Salkaházi Sára szűz és vértanúnak*.

tukusejssirs replied:
Thank you for trying to find a solution to the issue! :pray:

As for the orthography rules, well, there has been (and probably are) some arguments between linguists and liturgists in Slovakia regarding a comma between a title like _Martyr_ and the following _and_: linguists say it is redundant, liturgists still use in (in both official liturgical books like the Roman Missal and in _Ordos_). I presume a similar issue might be in Hungarian too.

At first, I tried to use grammatically correct celebration names, later I shifted to use celebration names from official/primary sources when available, as I (now) consider the celebration names officially _assigned_ and thus it should not be changed (unless it contains a typo or the celebration name is officially changed, e.g. a title added or canonisation level changed to `Saint`). For celebrations that are not part of any calendar proper in a particular language (based on the region/country in which a particular language is officially used), I try to translate the celebration name in the same way as those celebrations that are officially translated.

Therefore I suggest to use celebration names as they are printed in the Hungarian translation of the Roman Missal as the first and primary source. If it uses commas between a personal name and its title(s), then use it. If not, don’t use it. If the Roman Missal sometimes use it and sometimes not, try to create a rule when to use it and when not, tell us about your findings (in order to let us document it) and abide those rules in all celebration names. :wink:

Just a note: that [_Direktórium 2022_](http://tar.liturgia.hu/Naptar/DIR2022_JAV_2021-12-13.pdf) contains many celebrations that are celebrated in Slovakia, but not in Hungary, which suggests the its editor (compiler) comes from or is somehow related to Slovakia, thus they does not necessarily know Hungarian perfectly (I might be wrong though), therefore they might made some mistakes when translating a Slovak celebration name (`Blahoslavenej Sáry Salkaháziovej, panny a mučenice`) to Hungarian (`Boldog Salkaházi Sára, szűz és vértanú`).

In the end, I think that in Hungarian the comma should not be used actually. [Here](https://www.salkahazisara.com/hu/liturgia-2/) this celebration is written without a comma between the personal name and the titles.

Thank you, again, for you time and willingness to help and improve romcal! :pray:

MMate2007 replied:
You're very welcome!

MMate2007 committed:
Fix mistakes

tukusejssirs commented on lib/locales/hu.ts:
```diff
@@ -148,7 +148,7 @@ export const locale: Locale = {
     //'205_blessed_martyrs_of_japan': '205 Blessed Martyrs of Japan',
     adalbert_of_prague_bishop: 'Szent Adalbert püspök és vértanú',
-    adalbert_of_prague_bishop_patron_of_poland: 'Szent Adalbert püspök és vértanú, Legyelország főpátrónája',
+    adalbert_of_prague_bishop_patron_of_poland: 'Szent Adalbert, püspök és vértanú, Legyelország főpátrónája',
```
I think the comma after `Adalbert` should be removed. It is not present in [_Direktórium 2022_](http://tar.liturgia.hu/Naptar/DIR2022_JAV_2021-12-13.pdf#page=97) and according to [what you said](https://github.com/romcal/romcal/pull/403#discussion_r1018231279) (based on the Hungarian orthography rules) it should be removed.

MMate2007 replied:
Yes, I put some commas into the names and then I thought about it and I deleted them but I forgot to correct this one. 😄

MMate2007 committed:
Fix mistake

tukusejssirs commented:
@MMate2007, no need to re-request my review on this PR until it is finished. As I see it, it is not yet finished, thus I don’t think we should move it from the WIP/draft state. For some reason, this PR was (twice) changed to _ready to merge_ (non-WIP) state. I presume it might be connected to reviews and/or review requests, however, I have no idea.

IMHO at the very least we should have localised all celebrations used in the Hungarian Proper (i.e. all national celebrations should be localised, possible even diocesan celebrations) in order to consider this PR a finished one. However, when you think you won’t have much time to work on this, just tell us, I might continue the work (albeit I have no access to any liturgical books ATM, although I might gain some access if necessary).

MMate2007 commented:
OK, I'll work on it.

MMate2007 committed:
Update hu.ts

MMate2007 committed:
Update hu.ts

tukusejssirs commented on lib/locales/hu.ts:
```diff
@@ -0,0 +1,938 @@
+  },
+
+  ranks: {
+    //solemnity: 'solemnity',
```
```suggestion
    solemnity: 'főünnep',
```

tukusejssirs commented on lib/locales/hu.ts:
```diff
@@ -186,8 +186,8 @@ export const locale: Locale = {
     andrew_apostle: 'Szent András apostol',
-    andrew_apostle_patron_of_russia: 'Szent András apostol, Oroszország patrónusa',
-    andrew_apostle_patron_of_scotland: 'Szent András apostol, Skócia patrónusa',
+    andrew_apostle_patron_of_russia: 'Szent András apostol, Oroszország pátrónája',
+    andrew_apostle_patron_of_scotland: 'Szent András apostol, Skócia pátrónája',
```
In the Hungarian translation of the Roman Missal (1991; ed. typ. 2), `[principal] patron`  is localised as `fővédőszent` for Saint Benedict of Nursia and Saint Stephen of Hungary, but `Főpátronája` for Blessed Virgin Mary, Patron of Hungary (`Szűz Mária, Magyarok Nagyasszonya, Magyarország Főpátronája`). Is there any difference in Hungarian in the meaning or usage of `főpátrona` and `fővédőszent`? As I understand it, `főpátrona` (literary _main patron_) is used for patronages and `fővédőszent` (literary _main protector saint [person]_) for saints that we pray for some kind of protection, however, in English and Slovak, we use a single (base) word for both: `patron`/`patrón`.

`társvédőszentjei` (`co-patrons`) is used for Saint Cyril and Saint Methodius. That same title is used for Saint Maurus, Bishop, who is a co-patron of the Diocese of Pécs.

`mr_hu_1991_ed2` also uses term `védőszent` (for patrons of all patrons of Hungarian archdioceses and dioceses, except of Saint Maurus mentioned above who is a co-patron). I have no idea (yet) when we should use which (`főpátronája`/`fővédőszent`/`védőszent` for principal patron; `védőszent`/`társvédőszent` for secondary patron).

MMate2007 replied:
I don't know if there is any difference between _főpátróna_ and _fővédőszent_, but I don't think. I'll ask priest about it.

MMate2007 replied:
Yes, they are the same.

tukusejssirs replied:
> Yes, they are the same.

What is the source for this statement? A word of a priest or a liturgist or a linguist?

I have no idea if there is a difference in their meaning or usage, however, IMHO we should strive to keep the celebration names in sync with the latest version of `mr_hu` and other sources, as well as we need to keep the names consistent (therefore we need to create some l10n rules, both general ones and per locale).

tukusejssirs reviewed (changes requested):
I have asked a priest (friend of mine) to take [images of the GRC](https://drive.google.com/drive/folders/1T7IvMccJAsfea21fnpT6gIhxHHdpLHjx?usp=share_link) from the Hungarian translation of the Roman Missal. Its latest edition is based on _editio typica altera_ (second typical edition) and it was published in 1991. Note that the images are far from perfect (the priest is not a technical person), however, they are sufficient for our use case. From now on, I reference this edition as `mr_hu_1991_ed2`.

I have [pre-processed the Calendar images](https://drive.google.com/drive/folders/1KaFnAJVXNY6f_XmVsoKmeHzbUfe8J2vv?usp=share_link), [OCR’ed it and proofed the text](https://drive.google.com/drive/folders/1JyksZqp3JXfrANiGxAGUmRhTV4DHhHBA?usp=share_link), and [created a JSON](https://drive.google.com/drive/folders/1vcCaugmDT4QokdHn0S4jYl_ki3QKJUzr?usp=share_link) ([using a custom-made, 99-line script](https://drive.google.com/drive/folders/1vcCaugmDT4QokdHn0S4jYl_ki3QKJUzr?usp=share_link)).

Here are the rules I abide when I proofed the OCR’ed text:

- I proofed it, however, I might make some mistakes; do check it on your own from the original or pre-processed images;
- I used sentence case for celebration names, i.e. only those letters are in uppercase that have to be:
   - I am not 100 % sure about solemnities (which are in all-caps in the `mr_hu_1991_ed2`) if I changed the case correctly; do proof it please;
- each celebration is on a single line;
- as for celebration ranks:
   - I chose to put them right after the celebration names, but I put a semicolon and a space between them;
   - I kept the uppercase initial letter for the rank names, although I think we could write them in lowercase in romcal;
- if multiple optional celebrations are inscribed on a particular day, I simply repeated the day of month number before each celebration name;
- albeit it is not related to the localisation, you can use the asterisks (`*`) to get the list of celebrations inscribed in the Hungarian (national or diocesan) Calendar Proper:
   - do, however, note that some celebrations from the GRC (those that are without an asterisk) might contain some changes too (like additional titles, date change, rank lowering/raising) and we want all of these changes to be included in `lib/particular-calendars/hungary.ts`;
   - also note that some celebrations might be added/changed/removed after the publication the `mr_hu_1991_ed2`;
- I added `*` to celebrations that are 100 % part of the Hungarian national/diocesan Calendar Proper, but are missing an asterisk:
   - for example: `Esztergom: A főszékesegyház felszentelése` (`The Dedication of the Primatial Cathedral Basilica of the Assumption of the Blessed Virgin Mary and Saint Adalbert, Esztergom, Hungary`);
- report any issues with the OCR’ed text (e.g. using this PR or writing me an email);
- as for English translation of the cathedral dedications, I’d prefer to translate it as `Dedication of the {Cathedral|Basilica} of {Title|Patronage}, {City}, Hungary`:
   - the celebration names used in `mr_hu_1991_ed2` is ambiguous, as they all are `Dedication of the Cathedral`, without telling us what is the patron of the cathedral nor in what city the cathedral is (although that is somewhat understandable from the diocese name);
- I am not sure how could be `innocent_xi_pope` celebrated as `memorial` alongside of `pontian_i_pope_and_hippolytus_of_rome_priest` celebrated as `optional memorial`, both on 13 August:
   - one thing I am sure that `pontian_i_pope_and_hippolytus_of_rome_priest` is today celebrated on 16 August, according to Hungarian [Ordo 2022](http://tar.liturgia.hu/Naptar/DIR2022_JAV_2021-12-13.pdf#page=141);
   - I have kept this as it is printed in `mr_hu_1991_ed2`;
   - same is true of `bernard_of_clairvaux_abbot` (`memorial`) and `john_eudes_priest` (`optional memorial`) on 19 August, however, after some time, `john_eudes_priest` seems to be moved to 18 August, according to [Ordo 2022](http://tar.liturgia.hu/Naptar/DIR2022_JAV_2021-12-13.pdf#page=142);
- `Szűz Mária mennybevétele, (Nagyboldogasszony)` (`assumption_of_the_blessed_virgin_mary`): IMHO the comma should be removed, as it is written for the Dioceses of Győr and Székesfehérvár, thus I removed the comma;
- I also found [`mr_hu_1991_ed2`](http://tar.liturgia.hu/Dokumentumok/Romai%20Misekonyv/MISSALE%20ROMANUM%201991.%20-%20hun.pdf) online (it can be also found [here](https://drive.google.com/drive/folders/1OCI5uK0h4J19rXp8v0U8nrqNQU5rFLsH)), but it does not contain the Calendar, however, you can check the celebration names in _Szentek saját miséi_ (from real page 535 / [PDF page 454](http://tar.liturgia.hu/Dokumentumok/Romai Misekonyv/MISSALE ROMANUM 1991. - hun.pdf#page=454)) and in _Időszaki rész_ (from real page 105 / [PDF page 24](http://tar.liturgia.hu/Dokumentumok/Romai%20Misekonyv/MISSALE%20ROMANUM%201991.%20-%20hun.pdf#page=24));
- I added `solemnity` rank for All Souls, as it is ranked as such, although it is usually omitted (not printed);
- I replaced `Kȩty` (with `ȩ`, `e` with cedilla) with `Kęty` (with `ę`, `e` with ogonek) in the name of Saint John of Kanty, as was Polish and in Polish, there is no `ȩ` letter, only `ę`, and `Kęty` is the correct spelling;
- you could add `// src: mr_hu_1991_ed2` comment after all localisation strings of celebration names sourced from the Roman Misal, for example:

   ```ts
   epiphany_of_the_lord: 'Urunk megjelenése (Vízkereszt)', // src: mr_hu_1991_ed2
   ```

- Hungarian localisation rules (based on `mr_hu_1991_ed2`; we should add these to `docs/`, but it should not be part of this PR):
   - titles are written in lowercase unless they describe Our Lord or Our Lady (then their first letter is in uppercase);
   - ranks are in lowercase;
   - canonisation levels are always written with uppercase first letter;
   - `főpátron` is used for `principal patron`;
   - `pátron` is used for `principal patron`;
   - `and`, `companions` and other auxiliary words (words not describing canonisation level, name, title, rank) are written in lowercase.

MMate2007 committed:
Update lib/locales/hu.ts

Co-authored-by: Tukusej’s Sirs <tukusejssirs@protonmail.com>

MMate2007 commented:
Can I ask what OCR software did you use? Just if I want to OCR something then I know this is a good software for Hungarian text.
Soon, a new Hungarian Roman Missal will be printed, and there are a lot of celebrations that aren't in the current Roman Missal (or the one on the photos).

tukusejssirs commented:
> Can I ask what OCR software did you use?

Tesseract. I always preprocess the images using Scantailor Advanced.

Note, however, that the output text quality depends on the input images quality. I always prefer scanning documents/books using a flatbed scanner (if possible) at 600 × 600 px resolution, otherwise I take pictures with my phone at the best light conditions possible (light room with ideally no shadows on the paper) and I place the camera in the middle of the page.

Also note that one always needs to proof (read) the documents, as I haven’t used a perfect OCR yet. YMMV

> Soon, a new Hungarian Roman Missal will be printed, and there are a lot of celebrations that aren't in the current Roman Missal (or the one on the photos).

Yes, I’ve read about it. When it comes out, we need to update the strings.

> The JSON file will be included or it is just a help for us?

The JSON files are intended to aid our (your?) work on romcal. Anyone can freely use it for any purposes.

emagnier commented:
Would it be possible to merge this initial work stage, even if there are still some points to verify or improve? Or do we try to have at least the Temporale strings localized (the seasons and periods)? What do you think @tukusejssirs? 

@MMate2007, do you plan to continue this work soon?
In any case, the addition of this new language is a nice quality enhancement for Romcal, and we appreciate your big work on it!! 🎉 

As for the remaining points to improve, what do you think about gathering them within an Issue? 

Merging this first work (even at an incomplete stage) might open up new uses of Romcal in Hungarian and could encourage new contributors to contribute in this language as well 😄  (@MMate2007, if you wish to continue contributing, you are more than welcome 😊).

MMate2007 commented:
> @MMate2007, do you plan to continue this work soon?
Yes, I'd to work on this project but I don't have a lot of time so I can't make regular changes. But if I have the time, I'll contribute.

MMate2007 committed:
Update hu.ts

tukusejssirs commented:
@emagnier, give me 24h to re-review the PR, then I’ll leave a comment. I’d like to localise all strings from `en` to `hu`, however, there are some other romcal tasks/issues I want to work on first.

tukusejssirs commented on lib/locales/hu.ts:
```diff
@@ -622,9 +622,9 @@ export const locale: Locale = {
-    margaret_of_antioch_virgin: 'Saint Margaret of Antioch, Virgin and Martyr',
-    margaret_of_hungary_religious: 'Saint Margaret of Hungary, Religious',
+    margaret_mary_alacoque_virgin: 'Saint Margaret Mary Alacoque, Virgin',*/
+    margaret_of_antioch_virgin: 'Antiokheai Szent Margit, szűz és vértanú',
```
@MMate2007, where do you found this `Antiokheai` word? It seems to be incorrect.

On 17 Oct, we celebrate Saint Ignatius of Antioch which is translated as `Antiochiai Szent Ignác püspök és vértanú` in both the [Roman Missal](https://drive.google.com/file/d/1vnk7PSgzqn9Kx3yYpGTJK0DgpdinBQtW/view) (ed typ 2) and the [Ordo 2022](https://tar.liturgia.hu/Naptar/DIR2022_JAV_2021-12-13.pdf#page=166).

Also, in Hungarian, _usually_ there is no comma before the titles (but there is before _Magyarország fővédőszentje_ after `Szent István`; as I see it, the comma is used before patronage titles, but not before ‘regular’ titles).

```suggestion
    margaret_of_antioch_virgin: 'Antiochiai Szent Margit szűz és vértanú',
```

tukusejssirs commented:
@emagnier, I have looked into this PR and I want to add all celebration names from the Hungarian RM (ed typ 2). I need some time to do this. I presume I could make by Monday, maybe sooner. Then, we could merge this PR. All the rest keys (celebrations not celebrated in Hungary) will require much more time, therefore, it could be done in a separate PR.

Do you have any objections?

tukusejssirs committed:
feat(hu): add Hungarian locale

This PR adds initial Hungarian localisation based especially on the Hungarian Roman Missal (1991, editio typica altera), secondarily on the Hungarian [_Ordo 2022_](https://tar.liturgia.hu/Naptar/DIR2022_JAV_2021-12-13.pdf).

Only those celebrations names were localised which are celebration in the national Hungarian calendar, however, no missing celebrations were added (those need to be added in a follow-up PR). No celebrations celebrated in dioceses only were added.

fix #248

tukusejssirs commented:
@emagnier, @MMate2007, I have pushed a commit to the repo (@MMate2007, don’t forget to pull the changes before you do anything with your `hungarian-translate` branch). I have added 49 localised celebrations (plus some those where patronage titles were added) out of 253 (all of them are sourced from RM ed typ 2).

Some notes:

- I am not sure whether `patrick_of_ireland_bishop_patron_of_ireland` should be `védőszentje` or rather `fővédőszentje`, just like `stephen_i_of_hungary` and `benedict_of_nursia_abbot_patron_of_europe`.
- `A győri Könnyező Szűz Mária` (literally: `Weeping Virgin Mary of Győr`): [here](https://www.queenofireland.net/) I can find that in English, they call her `Our Lady Queen of Ireland, Consoler of the Afflicted`. I am not sure how to localise this celebration name, therefore I have skipped its addition for now (it could be added before merging this PR, however, as it is not yet used, it could be added in a separate PR in which we add the diocesan calendar of Győr).
- I am not sure whether `joseph_spouse_of_mary_patron_of_canada` should be `védőszentje` or rather `fővédőszentje`, however, I presume that this patron is a secondary patron and St Anne is the principal patron of Canada.

I won’t work on this tonight anymore, I’ll probably revisit this PR tomorrow evening CEST. :yawning_face: :wave:

emagnier commented:
@tukusejssirs absolutely no pressure. I just wanted to follow up here and get an update on the status. 

If time is limited, I'd suggest considering breaking down the work into multiple iterations (meaning multiple PRs). This could help in keeping PRs from being open for too long, and also ensure that the work already done can benefit Hungarian speakers and potentially attract new contributors in that language.

If you have the possibility to contribute here as well, that's really great! However, translating all the keys into `en` is not a priority from my point of view (at a minimum, translate the keys related to `seasons`, and ideally the keys used by the national Hungarian calendar, as you were mentioning).

tukusejssirs commented:
> @tukusejssirs absolutely no pressure. I just wanted to follow up here and get an update on the status.
>
> If time is limited, I'd suggest considering breaking down the work into multiple iterations (meaning multiple PRs). This could help in keeping PRs from being open for too long, and also ensure that the work already done can benefit Hungarian speakers and potentially attract new contributors in that language.

👍

> If you have the possibility to contribute here as well, that's really great! However, translating all the keys into `en` is not a priority from my point of view (at a minimum, translate the keys related to `seasons`, and ideally the keys used by the national Hungarian calendar, as you were mentioning).

Okay, how about adding a new locale in steps like this (provided, we uave the sources available, esp the vernacular RM):
- localise all regional/national celebrations listed in the calendar table of the Missal, and `seasons`;
- localise all celebrations celebrated only in dioceses/cities, but still listed in the calendar table of the Missal;
- localise the remaining celebrations from `en`.

---

Regarding `seasons`, I need to get my hands on the Roman Missal (or ask a friend of mine to take some pictures for me; or maybe @MMate2007 can help me out). Yes, I could use some online sources, but I trust them less than the RM. :wink:

emagnier commented:
@tukusejssirs as I said, absolutely no pressure. If you can provide updates within a week, that's great! If it takes more time, that's perfectly fine too 👍  And it's great that you can access Hungarian sources!

My previous comment was simply to get a status about the work from this PR.
I just want to avoid these types of PRs (calendar definitions & translations) remaining open for too long without any status updates. 

So, if we don't have time to complete everything we initially planned, it makes sense to merge the work already done, as long as it doesn't introduce regressions (on the existing code and data) and adds value (even if translations are still incomplete). 

And what's merged will be easier to maintain alongside the rest of the data and codebase.

In our case here, we can therefore wait a bit, since you plan to work on it soon 😄

MMate2007 committed:
Update hu.ts

tukusejssirs committed:
feat(hu): add Hungarian locale

This PR adds initial Hungarian localisation based especially on the Hungarian Roman Missal (1991, editio typica altera), secondarily on the Hungarian [_Ordo 2022_](https://tar.liturgia.hu/Naptar/DIR2022_JAV_2021-12-13.pdf).

Only those celebrations names were localised which are celebration in the national Hungarian calendar, however, no missing celebrations were added (those need to be added in a follow-up PR). No celebrations celebrated in dioceses only were added.

fix #248

tukusejssirs commented:
@emagnier, @MMate2007, I have pushed a commit to the repo (@MMate2007, don’t forget to pull the changes before you do anything with your `hungarian-translate` branch). I have added 49 localised celebrations (plus some those where patronage titles were added) out of 253 (all of them are sourced from RM ed typ 2).

Some notes:

- I am not sure whether `patrick_of_ireland_bishop_patron_of_ireland` should be `védőszentje` or rather `fővédőszentje`, just like `stephen_i_of_hungary` and `benedict_of_nursia_abbot_patron_of_europe`.
- `A győri Könnyező Szűz Mária` (literally: `Weeping Virgin Mary of Győr`): [here](https://www.queenofireland.net/) I can find that in English, they call her `Our Lady Queen of Ireland, Consoler of the Afflicted`. I am not sure how to localise this celebration name, therefore I have skipped its addition for now (it could be added before merging this PR, however, as it is not yet used, it could be added in a separate PR in which we add the diocesan calendar of Győr).
- I am not sure whether `joseph_spouse_of_mary_patron_of_canada` should be `védőszentje` or rather `fővédőszentje`, however, I presume that this patron is a secondary patron and St Anne is the principal patron of Canada.



---

@emagnier, @MMate2007, I have just finished localising all celebrations from `mr_hu_1991_ed2` (in `.names` only), except for:

- the missing national celebrations:
  - these celebrations will be added in a follow-up PR where we update `lib/particular-calendars/hungary.ts`;
- celebrations celebrated in the dioceses only;
  - again, these celebrations will be added in a follow-up PR where we update `lib/particular-calendars/hungary.$diocese_name.ts`.

Some changes which I made:

- `mr_hu_1991_ed2` uses `védőszentje` for `[principal] patron` and `társvédőszentje` for `[secondary/co-] patron` (`[fő-] pátrónája` is not used by `mr_hu_1991_ed2` at all):
  - there are some exceptions to this rule (these use `fővédőszentje`):
    - `benedict_of_nursia_abbot_patron_of_europe`;
    - `stephen_i_of_hungary_patron_of_hungary` (to be added in a follow-up PR to update `lib/particular-calendars/hungary.ts`).


Here’s a list of tasks that should be done before merging this PR:

- [ ] localise seasonal keys;
- [ ] `hungarian_saints_and_blesseds`: consider renaming the ID and English value to `All Saints of Hungary`;
- [ ] verify all l10n key values using `mr_2023_hu_ed3`, esp:
  - [ ] `our_lord_jesus_christ_the_eternal_high_priest`: current source for the translation is a second-party website: (liturgia.hu);
  - [ ] `stephen_the_first_martyr`: is there `diakónus` title used? it is used by [_Ordo 2022_](https://tar.liturgia.hu/Naptar/DIR2022_JAV_2021-12-13.pdf#page=42).
  - [ ] `gisela_of_hungary`: is there `királyné` title used?
  - [ ] `adalbert_of_prague_bishop_patron_of_poland` (if it is listed in the Missal): `főpátrónája` or `védőszentje`?
  - [ ] `matthias_apostle`: does he have `Evangelist` title;
  - [ ] `dedication_of_the_basilicas_of_saints_peter_and_paul_apostles`: in `mr_hu_1991_ed2`, there is no initial definite article (`A`), however, in [_Ordo 2022_](https://tar.liturgia.hu/Naptar/DIR2022_JAV_2021-12-13.pdf#page=178), there is;
- [ ] ask @MMate2007 for a review:
  - [ ] I am not sure whether `joseph_spouse_of_mary_patron_of_canada` should be `védőszentje` (`patron`) or rather `fővédőszentje` (`co-patron`), however, AFAIK this patron is a secondary patron and St Anne is the principal patron of Canada;
  - [ ] `denis_of_paris_bishop_patron_of_the_archdiocese_of_paris` and `denis_of_paris_bishop_patron_of_the_city_and_of_the_diocese_of_saint_denis`: verify my translations of the patronage titles;
- [ ] ask @emagnier for a review;
- [ ] rebase the PR branch on top of `dev` (if necessary);
- [ ] consider merging some/all commits;
- [ ] run `pnpm run lint:fix`;
- [ ] run `pnpm run prettier:fix`;
- [ ] run `pnpm run test:snapshot:update`;
- [ ] ??? remove untranslated strings which are commented out:
  - I suggest against doing this, as this is helpful to me for the future localisation work;

Create an issue for these:
 
- update national calendar of Hungary;
- add Hungarian diocesan calendars:
  - [Diocese of Győr] `A győri Könnyező Szűz Mária` (literally: `Weeping Virgin Mary of Győr`): [here](https://www.queenofireland.net/) I can find that in English, they call her `Our Lady Queen of Ireland, Consoler of the Afflicted`. I am not sure how to localise this celebration name;
- localise missing keys in `hu`.

MMate2007 committed:
Correct grammar

MMate2007 committed:
Correct grammar

MMate2007 commented:
> I am not sure whether joseph_spouse_of_mary_patron_of_canada should be védőszentje (patron) or rather fővédőszentje (co-patron), however, AFAIK this patron is a secondary patron and St Anne is the principal patron of Canada

If there are more than one patrons and St Joseph isn't the primary patron, then I think it should be _védőszentje_.

> denis_of_paris_bishop_patron_of_the_archdiocese_of_paris and denis_of_paris_bishop_patron_of_the_city_and_of_the_diocese_of_saint_denis: verify my translations of the patronage titles;

I've made some minor changes there.

> our_lord_jesus_christ_the_eternal_high_priest: current source for the translation is a second-party website: (liturgia.hu);

Actually, liturgia.hu isn't really a second-party website, they make the Direktórium, which is the official liturgical calendar here in Hungary.

tukusejssirs commented on lib/locales/hu.ts:
```diff
@@ -337,7 +337,7 @@ export const locale: Locale = {
     // demetrius_of_thessaloniki_martyr: 'Saint Demetrius of Thessaloniki, Martyr',
     denis_of_paris_bishop_and_companions_martyrs: 'Szent Dénes püspök és társai vértanúk',  // src: mr_hu_1991_ed2
-    denis_of_paris_bishop_patron_of_the_archdiocese_of_paris: 'Szent Dénes püspök és társai vértanúk, a Párizsi főegyházmegye védőszentje',  // based on: mr_hu_1991_ed2
+    denis_of_paris_bishop_patron_of_the_archdiocese_of_paris: 'Szent Dénes püspök és társai vértanúk, a párizsi főegyházmegye védőszentje',  // based on: mr_hu_1991_ed2
```
I know that adjectives of proper names are written with lower-case initial letter, however, as I understood it, _Párizsi főegyházmegye_ is a proper name in itself, therefore it should be written with a capital `P`. If the contrary would be the case, then `Saint-Denis-i` in `denis_of_paris_bishop_patron_of_the_city_and_of_the_diocese_of_saint_denis` would be lower-cased too, don’t you think? :thinking:

Could you verify this statement of mine? I might be wrong, as I have never learnt Hungarian at school. :thinking:

Thanks! :pray:

MMate2007 replied:
Sorry, I didn't think that _párizsi_ is the actual name, I just automatically corrected it. It's my mistake.

tukusejssirs replied:
Okay, so it needs to be reverted.

tukusejssirs commented:
> If there are more than one patrons and St Joseph isn't the primary patron, then I think it should be _védőszentje_.

Well, according to `mr_hu_1991_ed2` which I have analysed, `védőszent` is used for all `[principal] patron` saints except for these two celebrations which use `fővédőszent` (`[principal] patron`):
- St Benedict of Nursia ([Principal] Patron of Europe);
- St Stephen of Hungary ([Principal] Patron of Hungary and of the Archdiocese of Kalacsa).

Then the following saints use `társvédőszent` (`copatron`):

- St Catherine of Siena (Copatroness of Europe);
- Sts Cyril and Methodius (Copatrons of Europe);
- St Maurice of Pécs (Copatron of the Archdiocese of Pécs).

Finally, all other patrons use `védőszent` (`patron`):

- St Adalbert of Prague (Patron of the Archdiocese of Esztergom);
- St Ladislaus of Hungary (Patron of the Northern part of the Diocese of Szeged-Csanád):
  - I am not sure if `Északi rész` (the Northern part) is a separate diocese or is it only the Northen part of the Diocese;
- St Peter, Apostle (Patron of the Diocese of Pécs);
- St Anne, Mother of the Blessed Virgin Mary (Patron of the Diocese of Veszprém);
- Assumption of the Blessed Virgin Mary (Patron of the Dioceses of Győr and Székesfehérvár);
- St Gerard of Csanád (Patron of the Diocese of Szeged-Csanád);
- St Micheal, Archangel (Patron of the Diocese of Vác);
- St Martin of Tours (Patron of the Diocese of Szombathely);
- St John, Apostle and Evangelist (Patron of the Archdiocese of Eger).

Based on this analysis, I assumed that `fővédőszent` and `védőszent` are the same title (`[principal] patron`) and `társvédőszent` is used for `copatron`.

Now, the question is when to use `fővédőszent` and when `védőszent`? What is the rule? Note that I presume that `mr_hu_2023_ed3` will use only one word for `[principal] patrons` and one for `copatrons`. There are multiple options:

| Principal Patron | Secondary Patron |
|------------------|------------------|
| `fővédőszent`    | `társvédőszent`  |
| `fővédőszent`    | `védőszent`      |
| `védőszent`      | `társvédőszent`  |

---

Therefore, it is not that easy to answer what patronage title to use for St Joseph. :thinking:

> > denis_of_paris_bishop_patron_of_the_archdiocese_of_paris and denis_of_paris_bishop_patron_of_the_city_and_of_the_diocese_of_saint_denis: verify my translations of the patronage titles;
> 
> I've made some minor changes there.

Thanks! I missed these typos. :pray:

> Actually, liturgia.hu isn't really a second-party website, they make the Direktórium, which is the official liturgical calendar here in Hungary.

In my personal definition, I divide the sources to three categories:

- first-party sources:
  - the official liturgical books, either in original (Latin) or in vernacular translations (e.g. the Roman Missal, the Liturgy of the Hours);
  - in this category I also count the decrees published by the Holy See:
    - these decrees are usually re-published in [_Acta Apostolicæ Sedis_](https://www.vatican.va/archive/aas/index_sp.htm), therefore those are also considered as the first-party sources;
  - some information might be also published in [_Notitiæ_](https://www.vatican.va/roman_curia/congregations/ccdds/documents/rc_con_ccdds_notitiae_index_en.html);
  - of course, translations of the decrees and _Notitiæ_ are considered as the first-party sources, regardless if the translation is official or not;
- second-party sources:
  - other publications from trusted authors/institutions (esp the Holy See, and episcopal conferences and their commisions);
  - for example:
    - ordos/directoriums;
    - official websites of the trusted authors/institutions;
- third-party sources:
  - anything else (e.g. Wikipedia).

Therefore, accroding to this categorisation, it is a second-party source.

tukusejssirs committed:
feat(hu): add Hungarian locale

This PR adds initial Hungarian localisation based especially on the Hungarian Roman Missal (1991, editio typica altera), secondarily on the Hungarian [_Ordo >

Only those celebrations names were localised which are celebration in the national Hungarian calendar, however, no missing celebrations were added (those nee>

fix #248

tukusejssirs commented:
I have localised all the seasonal strings according to the sources available to me. They should be, however, still verified (esp using `mr_hu_2023_ed3`) in order to make sure they are correctly localised.

---

I have added the following tasks to [this comment of mine](https://github.com/romcal/romcal/pull/403#issuecomment-1703894988). I won’t update the list here, but in that comment. I just want to notify you guys (esp you, @MMate2007) that they are added to that comment.
- verify with `mr_hu_2023_ed3`:
  - [ ] `.seasons.paschal_triduum.season`: I am not sure whether we should keep the initial definite artitle (`A`) or not:
    - it is printed in [_Ordo 2022_](https://tar.liturgia.hu/Naptar/DIR2022_JAV_2021-12-13.pdf#page=82), however, no other season has it;
  - [ ] I haven’t found a source for these keys, I simply translated the usual format used, e.g., in English and Slovak:
    - `.seasons.christmas_time.day` (I used `${weekDayName} karácsonyi időben`, e.g `Hétfő karácsonyi időben`);
    - `.seasons.christmas_time.before_epiphany` (I used `${monthName} ${dayInMonth}.`, e.g `December 17.`);
    - `.seasons.christmas_time.after_epiphany` (I used `${weekDayName} az Úrunk megjelenése után`, e.g `Hétfő az Úrunk megjelenése után`);
  - [ ] `.color.purple`: I have changed it to `viola` (as per [_Ordo 2022_](https://tar.liturgia.hu/Naptar/DIR2022_JAV_2021-12-13.pdf#page=172)), although [lh.kbs.sk/hu](https://lh.kbs.sk/cgi-bin/l.cgi?qt=pdt&d=4&m=12&r=2023&j=hu) uses `lila`;
  - [ ] `.color.gold`: I could not find a source for this colour, but it should be correct;
  - [ ] `.color.rose`: the colour name in [_Ordo 2022_](https://tar.liturgia.hu/Naptar/DIR2022_JAV_2021-12-13.pdf#page=35) is shortened to `rózs.`, but again, it should be correct;
  - [ ] `.cycles`:
    - [ ] `.cycles.proper_of_time` and `.cycles.proper_of_saints`:
      - I am not sure how to translate these keys;
      - I simply ‘hungarianised’ their Latin counterparts and provided some third-party sources for them;
      - I could imagine that `mr_hu_2023_ed3` translates them something like `A szentek saját része`, as it is in `mr_sk_2021_ed3`, however, I could not find a similar translation for the Proper of Time (`Az idők saját része`?);
    - [ ] all the other cycles:
      - I simply kept the translations of @MMate2007, as they seem to be correct, but sourcing them from the [_Ordo 2022_](https://tar.liturgia.hu/Naptar/DIR2022_JAV_2021-12-13.pdf) seems to me bad;
- [ ] test the romcal generation out using the `hu` locale;

tukusejssirs commented on lib/locales/hu.ts:
```diff
@@ -0,0 +1,936 @@
+  seasons: {
+    advent: {
+      season: 'Adventi idő',  // src: https://tar.liturgia.hu/Naptar/DIR2022_JAV_2021-12-13.pdf#page=29
+              weekday: 'Advent {{week, romanize}}. hete $t(weekdays:{{dow}})',  // src: https://kattima.hu/hu/app/pray/bences-zsolozsma-l-a-u-d-e-s-advent-ii.-hete-hetfo-80378
```
@MMate2007, this one sounds weird to me, however, I could not (yet) find a better source for this string.

MMate2007 replied:
Please give me an example of what it will display.

tukusejssirs replied:
`Advent I. hete hétfő`.

Explanation:
- `{{week}}` is replaced with the week number in Arabic numerals (e.g. `1`);
- `{{week, romanize}}` returns the week number in Roman numerals (e.g. `IV`);
- `{{dow}}` is replaced with the day of week number (`0` for Sundays, `6` for Saturdays);
- `$t(weekdays:{{dow}})` returns the localised day of week (e.g. `hétfő`):
  - see the `.weekdays` in `hu.ts`;
- `$t(weekdays:{{dow}}, capitalize)` returns the localised day of week with capital initial letter (e.g. `Hétfő`).

tukusejssirs commented on lib/locales/hu.ts:
```diff
@@ -0,0 +1,936 @@
+
+    ordinary_time: {
+      season: 'Évközi idő',  // src: https://tar.liturgia.hu/Naptar/DIR2022_JAV_2021-12-13.pdf#page=50
+      weekday: '$t(weekdays:{{dow}}, capitalize) évközi időben, {{week, romanize}}. hét',  // src: https://kattima.hu/hu/app/pray/bences-zsolozsma-k-o-m-p-l-e-t-o-r-i-u-m-csutortok-evkozi-idoben-i.-het-77548
```
@MMate2007, this one sounds weird to me, however, I could not (yet) find a better source for this string.

MMate2007 replied:
Please give me an example of what it will display.

tukusejssirs replied:
`Hétfő évközi időben, I. hét`.

See [this comment of mine](https://github.com/romcal/romcal/pull/403#discussion_r1316094197).

MMate2007 replied:
It seems right to me.

tukusejssirs replied:
Grammatically, yes, but is it actually used this way? In `hu_mr_2022_ed3` (which I have OCR’d and [optimised](https://drive.google.com/drive/u/3/folders/1F4UwlJ-ajcd_VfeBBiGgMJwKD2Kyj_7T), but forgot to tell you about it) does not use this celebration name in full (it is simply `Hétfő`/`Monday`), therefore this might need to be verified using other liturgical books (e.g. lectionary).

emagnier commented:
Can we rebase + merge at least this initial part of the work?

And is it possible to add these translations at least in this PR (with the help of a translation tool if necessary, unless @MMate2007 you can provide them for us, even simply in a comment here)?

```
WARN  Missing localized 'hu' items for the Proper of Time and the General Roman Calendar:
  ranks:
       - optional_memorial
  names:
       - easter_sunday
       - gregory_of_narek_abbot
       - paul_vi_pope
       - apollinaris_of_ravenna_bishop
       - peter_julian_eymard_priest
       - catherine_of_alexandria_virgin
```

This will allow us to make progress more smoothly on this translation work (having a first operational `hu` version in Romcal). And since we are currently doing some large refactoring on the project, it will prevent us from having to maintain this PR open any longer.

MMate2007 committed:
Add translation of names

easter_sunday and paul_vi_pope
